packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -199,30 +209,64 @@ export function PanelChrome({
   // Handle drag & selection events
   // Mainly the tricky bit of differentiating between dragging and selecting
 
-  const onPointerUp = (evt: React.PointerEvent) => {
-    evt.stopPropagation();
+  const resetPointerEventConfig = () => {
+    pointerEventConfig.current = { pointerDown: false, dragStarted: false, initialPosition: { x: 0, y: 0 } };
+  };
 
-    const distance = Math.hypot(
-      pointerDownEvt.current?.screenX ?? 0 - evt.screenX,
-      pointerDownEvt.current?.screenY ?? 0 - evt.screenY
-    );
+  const onPointerDown = (evt: React.PointerEvent) => {
+    // We ignore clicks on elements that should cancel dragging (panel menu, etc)
+    // Also reset the configuration
+    if (dragClassCancel && evt.target instanceof Element && evt.target.closest(`.${dragClassCancel}`)) {
+      resetPointerEventConfig();
+      return;
+    }
 
-    pointerDownEvt.current = undefined;
+    evt.stopPropagation();
+    pointerEventConfig.current = {
+      pointerDown: true,
+      dragStarted: false,
+      initialPosition: { x: evt.screenX, y: evt.screenY },
+    };
+  };
 
-    // If we are dragging some distance or clicking on elements that should cancel dragging (panel menu, etc)
+  // When the pointer is moved, we need to trigger drag event if the pointer has moved more than a threshold
+  const onPointerMove = (evt: React.PointerEvent) => {
+    if (!pointerEventConfig.current.pointerDown || pointerEventConfig.current.dragStarted) {
+      return;
+    }
+
+    evt.stopPropagation();
     if (
-      distance > 10 ||
-      (dragClassCancel && evt.target instanceof HTMLElement && evt.target.closest(`.${dragClassCancel}`))
+      Math.sqrt(
```

Comment:
I thought I had already implemented this logic 2 weeks ago , where did that go?

Also there is a math function that does this distance calculation

Reply:
It was replaced in the dnd PR. Left a comment back then.

The problem with `Math.hypot` is that on some browsers, Safari mainly, it's 2x worse in terms of performance.